websites/S/Shikimori/presence.ts

```diff
@@ -3,152 +3,206 @@ const presence = new Presence({
-			presenceData.state = "Пересматриваю";
-		else if (document.location.pathname.includes("/completed"))
-			presenceData.state = "Просмотрено";
-		else if (document.location.pathname.includes("/on_hold"))
-			presenceData.state = "Отложено";
-		else if (document.location.pathname.includes("/dropped"))
-			presenceData.state = "Брошено";
+	function isImageExist(tags: string) {
+		return document?.querySelector<HTMLImageElement>(tags) && !privacy && logo
+			? document?.querySelector<HTMLImageElement>(tags)?.src
+			: "https://i.imgur.com/7tlc7or.png";
+	}
+
+	if (buttons && !privacy) {
+		presenceData.buttons = [
+			{
+				label: "Открыть страницу",
+				url: document
+					.querySelector("meta[property='og:url']")
+					.getAttribute("content"),
+			},
+		];
+	}
 
-		presenceData.details = `Смотрит ${document
-			.querySelector(".submenu-triangle > span")
-			.textContent.toLowerCase()} ${document.location.pathname.split("/")[1]}`;
-		presenceData.largeImageKey = (<HTMLImageElement>(
-			document.querySelector(".block.avatar center a img")
-		)).src;
-	} else if (document.location.pathname === `/${title}`) {
-		presenceData.details = `В профиле ${title}`;
-		presenceData.largeImageKey = (<HTMLImageElement>(
-			document.querySelector(".avatar img")
-		)).src;
-	} else if (
-		document.location.pathname.includes("/list/anime/") ||
-		document.location.pathname.includes("/list/manga/")
-	) {
-		presenceData.details = `Смотрит ${document
-			.querySelector(".submenu-triangle > span")
-			.textContent.toLowerCase()} ${document.location.pathname.split("/")[1]}`;
-		presenceData.largeImageKey = (<HTMLImageElement>(
-			document.querySelector(".block.avatar center a img")
-		)).src;
-	} else if (
-		document.location.pathname.includes("/characters/") ||
-		document.location.pathname.includes("/people/")
-	) {
-		let type;
-		if (document.location.pathname.includes("/characters/")) type = "персонажа";
-		else if (document.location.pathname.includes("/people/")) type = "человека";
-		presenceData.largeImageKey = (<HTMLImageElement>(
-			document.querySelector(".c-poster center img")
-		)).src;
-		presenceData.details = `Смотрит страницу ${type}`;
-		presenceData.state = title;
+	switch (pathname.split("/")[1]) {
+		case "":
+			presenceData.details = "На главной странице";
+			break;
+		case "animes":
+		case "mangas":
+		case "ranobe":
+			presenceData.details = `В поисках ${
+				location.pathname.split("/")[1] === "animes"
+					? "аниме"
+					: pathname.split("/")[1] === "mangas"
+					? "манги"
+					: "ранобэ"
+			}`;
+			if (pathname.split("/")[2]) {
+				switch (pathname.split("/")[2]) {
+					case "genre":
+					case "kind":
+					case "status":
+					case "order-by":
+					case "mylist":
+					case "season":
+					case "score":
+					case "duration":
+					case "rating":
+					case "studio":
+						if (!privacy) {
+							presenceData.state =
+								document?.querySelector("header > h1")?.textContent;
```

Comment:
`document` should always be defined, so there is no need to use optional chaining on it.
```suggestion
								document.querySelector("header > h1")?.textContent;
```
(same for the other instances of this.)

Reply:
> `document` should always be defined, so there is no need to use optional chaining on it.
> 
> (same for the other instances of this.)

Oops, I don't know why I did it. Thanks.